### PR TITLE
 Add bindable for 'safe area' padding

### LINKED ISSUE #2075: Expose safe area insets for mobile platforms
Many new devices have rounded screen corners and/or a "notch" which need to be catered for when positioning UI elements.  The iPhone X* and new iPad Pro also have the home bar at the bottom of the screen.

I propose we expose those as part of the framework, or potentially at an osuTK level instead.

iOS:
https://developer.xamarin.com/api/property/UIKit.UIView.SafeAreaInsets/

Android:
https://developer.xamarin.com/api/type/Android.Views.WindowInsets/

Some more info on the safe areas on iOS and Android:
https://v-play.net/cross-platform-app-development/notch-developer-guide-ios-android

### PR DESCRIPTION
**Pull request by swoolcock**
Newer phones and tablets often have notches and rounded corners.  This PR exposes these as a bindable, since they dynamically change when the device is rotated.  Framework consumers can use this to layout UI elements so that they are always visible.

Includes tests, but the results are only evident when running on a device.  Can't really add any asserts either, so it's mostly for visual verification.

Will need to implement this in Android too.

- Closes #2075

---

Add support for 'safe area' padding on mobile devices.

### CODE REVIEW

File: osu.Framework.iOS/IOSGameView.cs [hunk 1]
```diff
@@ -45,6 +45,20 @@ protected override void ConfigureLayer(CAEAGLLayer eaglLayer)
 
         public float Scale { get; private set; }
 
+        // SafeAreaInsets is cached to prevent access outside the main thread
+        private UIEdgeInsets safeArea = UIEdgeInsets.Zero;
+        public UIEdgeInsets SafeArea
```
Comment (peppy) [inline osu.Framework.iOS/IOSGameView.cs]: is there a reason this is public?

File: osu.Framework/Platform/GameWindow.cs [hunk 2]
```diff
@@ -189,6 +189,8 @@ private string getVersionNumberSubstring(string version)
 
         protected virtual void OnKeyDown(object sender, KeyboardKeyEventArgs e) => KeyDown?.Invoke(sender, e);
 
+        public readonly BindableMarginPadding SafeAreaPadding = new BindableMarginPadding();
```
Comment (peppy) [inline osu.Framework/Platform/GameWindow.cs]: what is a safe area? xmldoc please

File: osu.Framework.iOS/IOSGameView.cs (hunk 1 above)
Comment (swoolcock) [reply osu.Framework.iOS/IOSGameView.cs]: Oversight, internal is enough.

**Comment by peppy** [comment]
I mean not a huge deal but the testcase for this is pretty pointless. The padding is added in a weird child location where it can't really be confirmed to be correct.

**Comment by swoolcock** [comment]
Yeah fair point, I just needed a way to visually verify on device that the padding was updating as it rotated.  It can be used to verify the Android implementation is working (once that's done).

**Pull request merged**